src/backend/marsha/core/permissions.py

```diff
@@ -271,6 +271,75 @@ def has_permission(self, request, view):
             return False
 
 
+class IsOrganizationAdmin(permissions.BasePermission):
+    """
+    Allow a request to proceed. Permission class.
+
+    Only if the user is a member of the organization in the path, with
+    an administrator role.
+    """
+
+    def has_permission(self, request, view):
+        """
+        Allow the request.
+
+        Only if the organization exists and the current logged in user is one
+        of its administrators.
+        """
+        return models.OrganizationAccess.objects.filter(
+            role=ADMINISTRATOR,
+            # Avoid making extra requests to get the organization id through get_object
+            organization__id=view.get_object_pk(),
+            user__id=request.user.id,
+        ).exists()
+
+
+class IsPlaylistAdmin(permissions.BasePermission):
+    """
+    Allow a request to proceed. Permission class.
+
+    Only if the user has an access to the playlist in the path, with
+    an administrator role.
+    """
+
+    def has_permission(self, request, view):
+        """
+        Allow the request.
+
+        Only if the playlist exists and the current logged in user is one
+        of its administrator.
+        """
+        return models.PlaylistAccess.objects.filter(
+            role=ADMINISTRATOR,
+            # Avoid making extra requests to get the playlist id through get_object
+            playlist__id=view.get_object_pk(),
+            user__id=request.user.id,
```

Comment:
```suggestion
            user=request.user,
```

Reply:
@sampaccoud we cannot do this as in Marsha API `request.user` is not a django user but just a token user, which does contain the actual user id.

What would work is `user=User.objects.get(id=request.user.id)` but we probably don't want to add a request here.